### DISCUSSION TITLE
half page & center cursor "c-d c-u + zz"

### DISCUSSION THREAD

**Discussion by darianmorat** [VSCodeVim/Vim#8809] [Q&A]
I couldn't find any way to remap `c-d` `c-u` with `zz` to center the page. When trying to combine on any way in the settings.json or keybindings.json is just bug and not working at all. It this possible in the extension? 

I was trying to get the change in the settings.json with:
```json
      {
         "before": ["c-d"],
         "after": ["M", "c-d"]
      },
      // or
      {
         "before": ["c-d"],
         "after": ["c-d", "zz"]
      },
```

any way to accomplish something like this?:
```vim
vim.keymap.set("n", "<C-u>", "<C-u>zz", { desc = "Half page up" })
vim.keymap.set("n", "<C-d>", "<C-d>zz", { desc = "Half page down" })
```

**Comment by pyrooka** [comment]
Hey there! I was looking for the same configuration and I've just found the solution. Each string in the array must be a single command or character to be "pressed" so you need to define `z` twice. Here is my config:
```json
    "vim.normalModeKeyBindings": [
        {
            "before": ["<C-d>"],
            "after": ["<C-d>", "z", "z"]
        },
        {
            "before": ["<C-u>"],
            "after": ["<C-u>", "z", "z"]
        }
    ],
```
I hope this helps! :)
